### ISSUE TITLE
v1.8.0: training and validation results on console screen are overwritten

### ISSUE THREAD

**Issue by LeMinhNgan** [roboflow/rf-detr#1127]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

I have observation that the console just shows training and validation result of epoch0, epoch1 and the last epoch. Result of other interim epochs were overwritten. We usually monitor these result summaries (loss, mAP50-95) to make decision whether stop or increase the max epochs or rework the dataset. Please consider bringing the training results of interim epochs back. Thanks!

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/921c4c80-51ad-4123-89af-411bb75a4e90" />

### Environment

rfdetr: 1.8.0
OS windows 11
RTX5090
python 3.10
pytorch 2.11

### Minimal Reproducible Example

```
model= RFDETRLarge(resolution=1120)
model.train(dataset_dir= cfg.dataset_dir, epochs=10, batch_size=2)
```

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!
